User_Story#1/SortingApp/SortinfApp.UnitTests/SortingAlgorithms/InsertionSorterTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Linq;
+using BusinessLayer.SortingAlgorithms;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SortingApp.UnitTests.SortingAlgorithms
+{
+    /// <summary>
+    /// Tests for InsertionSorter class.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class InsertionSorterTest
+    {
+        /// <summary>
+        /// Tests Sort method if pass null collection it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void InsertionSorter_Sort_PassNullCollection_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            double[] array = null;
+            var insertionSorter = new InsertionSorter<double>();
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => insertionSorter.Sort(array));
+        }
+
+        /// <summary>
+        /// Tests Sort method if pass sorted collection it will leave sorted collection.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void InsertionSorter_Sort_PassSortedCollection_GetSortedCollection()
```

Comment:
I believe this logic is not valid for our test case... but let it be.

Reply:
as a variant, I could read sorted array from file and I want to be sure that logic of my Sort method doesn't compromise the output.

Reply:
Agree.